### DISCUSSION TITLE
New pycaret 3.0.0rc3 dependencies conflicts

### DISCUSSION THREAD

**Discussion by Breno-BR** [pycaret/pycaret#2777] [Q&A] [answered]
I have a problem to import and use pycaret.time_series with Zorin OS 16 and Python version: 3.9.2.
The strange behavior is that everything went ok when running at my RaspberryPi 8Gb.

My system config:

OS: Zorin OS 16.1 x86_64 
Host: X470 AORUS ULTRA GAMING 
Kernel: 5.15.0-41-generic 
Shell: bash 5.0.17 
DE: GNOME 
WM: Mutter 
Terminal: gnome-terminal 
CPU: AMD Ryzen 5 2600 (12) @ 3.400GH 
GPU: NVIDIA GeForce GTX 1660 Ti 
Memory: 4678MiB / 32047MiB 

Installed with: 

`pip install --pre pycaret`

First error:

```
>>> from pycaret.time_series import *
runtimeerror: module compiled against api version 0x10 but this version of numpy is 0xe
traceback (most recent call last):
  file "<stdin>", line 1, in <module>
  file "/home/user/anaconda3/envs/carga/lib/python3.9/site-packages/pycaret/time_series/__init__.py", line 1, in <module>
    from pycaret.time_series.forecasting.functional import (
  file "/home/user/anaconda3/envs/carga/lib/python3.9/site-packages/pycaret/time_series/forecasting/functional.py", line 13, in <module>
    from pycaret.time_series.forecasting.oop import tsforecastingexperiment
  file "/home/user/anaconda3/envs/carga/lib/python3.9/site-packages/pycaret/time_series/forecasting/oop.py", line 15, in <module>
    from plotly_resampler import figureresampler, figurewidgetresampler
  file "/home/user/anaconda3/envs/carga/lib/python3.9/site-packages/plotly_resampler/__init__.py", line 3, in <module>
    from .aggregation import lttb, efficientlttb, everynthpoint
  file "/home/user/anaconda3/envs/carga/lib/python3.9/site-packages/plotly_resampler/aggregation/__init__.py", line 11, in <module>
    from .aggregators import (
  file "/home/user/anaconda3/envs/carga/lib/python3.9/site-packages/plotly_resampler/aggregation/aggregators.py", line 18, in <module>
    from .algorithms import lttbcv2
importerror: numpy.core.multiarray failed to import
```

Possible solution (upgrade numpy):

`pip install -U numpy`

New error:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behavior is the source of the following dependency conflicts.
sktime 0.11.4 requires numpy<1.22,>=1.21.0, but you have numpy 1.23.1 which is incompatible.
numba 0.55.2 requires numpy<1.23,>=1.18, but you have numpy 1.23.1 which is incompatible.
Successfully installed numpy-1.23.1
```

Now I got a conflict between numpy, numba and sktime.
Can anyone help?

Thanks in advance! :)

**Comment by ngupta23** [comment, marked as answer]
I do not think there is a conflict between numpy, numba and sktime since both numba and sktime can use numpy 1.21.0 per your error message. What happens when you install numpy version 1.21.0 manually? Can you try that?

**Reply by Breno-BR** [reply]
Thanks for your reply! :)

After pycaret install I got the first error.

When I update numpy to 1.23.1, the first error seems to disappear but then I got a conflict with numba and sktime.

Tried to change numpy to 1.21.0 then the first error got back:

`ImportError: numpy.core.multiarray failed to import`

:/

**Reply by ngupta23** [reply]
I think it is an OS specific issue. Is numpy installed at all? Can you install version 1.21.0 separately in a new environment?

**Reply by Breno-BR** [reply]
Maybe it is really an OS issue. I am already using a conda environment. Numpy is installed and returned multiarray error with both 1.21.0 and 1.21.6 versions. I will try another OS now and reply my results here. :)

**Reply by Breno-BR** [reply]
Everything is working with another SO:

> Edition	Windows 10 Pro
> Version	21H2
> SO compilation	19044.1826
> Experience	Windows Feature Experience Pack 120.2212.4170.0

Python version: **3.8.8**

Just one error on the installation process:

```
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for plotly-resampler
Failed to build plotly-resampler
ERROR: Could not build wheels for plotly-resampler, which is required to install pyproject.toml-based projects
```

To get it working just installed **plotly-resampler** before **pycaret**:

First:
`pip install plotly-resampler`

Then:
`pip install --pre pycaret`

Thanks for your help and congratulations on the excellent work! @ngupta23 

Best regards from Brazil! :)
